docs/plugin-deploy.md

```diff
@@ -18,18 +13,55 @@ $ ls
 main.go go.mod go.sum
 ```
 
-If you are in GOPATH, enable go modules explicitly:
+(Optional) Make sure there are no conflicting dependencies:
 
 ```bash
-$ export GO111MODULE=on
+$ go get -u github.com/gotify/plugin-api/cmd/gomod-cap
+$ go run github.com/gotify/plugin-api/cmd/gomod-cap \
+         -from /path/to/gotify/server/source/go.mod -to /path/to/plugin/source/go.mod
+$ go mod tidy
 ```
 
-(Optional) Make sure there are no conflicting dependencies:
+### With Docker (recommended)
+
+The [gotify/build docker images](https://github.com/gotify/build) are used for building gotify/server therefore it is
+it is recommended to use the same build environment for plugins to ensure compatibility.
+
+You can build your plugin like this:
+
+#### linux amd64
 
 ```bash
-$ go get -u github.com/gotify/plugin-api/cmd/gomod-cap
-$ go run github.com/gotify/plugin-api/cmd/gomod-cap -from /path/to/gotify/server/source/go.mod -to /path/to/plugin/source/go.mod
-$ go mod tidy
+$ docker run --rm -v "$PWD/.:/proj" -w /proj gotify/build:v1-linux-amd64 \
```

Comment:
How about tagging images with go version?

Reply:
Done in https://github.com/gotify/build/commit/43beb35a22fb67876e5a15cf7cbf473ee56de6e5